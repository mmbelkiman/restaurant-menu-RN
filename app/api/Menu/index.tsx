export interface IApiMenuItem {
  name: string;
  description: string;
  price: number;
  url: string;
}

export interface ILocation {
  lat: string;
  log: string;
  address1: string;
  address2: string;
  city: string;
  state: string;
  postalCode: string;
  country: string;
}

export interface IApiMenu {
  id: string;
  name: string;
  description: string;
  phone: string;
  location: ILocation;
  operationDays: [];
  privateParking: boolean;
  currency: string;
  menus: Array<IApiMenus>;
}

export interface IApiMenus {
  name: string;
  items: Array<IApiMenuItem>;
}

export const Menu = {
  get: (): Promise<IApiMenu> => {
    let requestOptions = {
      method: 'GET',
      redirect: 'follow',
    };

    return new Promise((resolve, reject) => {
      fetch('https://mcdonalds.trio.dev/menu', requestOptions)
        .then(response => response.json())
        .then((result: IApiMenu) => resolve(result))
        .catch(() => reject(undefined));
    });
  },
};
