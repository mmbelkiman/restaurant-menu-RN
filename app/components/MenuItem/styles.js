import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  container: {
    padding: 30,
  },
  name: {
    textAlign: 'center',
    fontSize: 23,
    color: 'black',
    fontFamily: 'PlayfairDisplay-Medium',
  },
  description: {
    textAlign: 'center',
    fontSize: 18,
    color: 'gray',
    paddingTop: 10,
  },
  price: {
    textAlign: 'center',
    fontSize: 14,
    color: 'gray',
    paddingTop: 10,
  },
});
