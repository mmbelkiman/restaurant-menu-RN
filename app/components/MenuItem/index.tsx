import React from 'react';
import {Text, View} from 'react-native';
import LineSeparator from '../LineSeparator';
import {IApiMenuItem} from '../../api/Menu';
import styles from './styles';

interface ILineSeparatorProps {
  item: IApiMenuItem;
  index: number;
}

const MenuItem: React.FC<ILineSeparatorProps> = ({item, index}) => {
  return (
    <View>
      <LineSeparator />

      <View style={styles.container}>
        <Text testID={`menu-item-title-${index}`} style={styles.name}>
          {item.name}
        </Text>
        <Text
          testID={`menu-item-description-${index}`}
          style={styles.description}>
          {item.description}
        </Text>
        <Text
          testID={`menu-item-price-${index}`}
          style={styles.price}>{`$ ${item.price}`}</Text>
      </View>
    </View>
  );
};

export default MenuItem;
