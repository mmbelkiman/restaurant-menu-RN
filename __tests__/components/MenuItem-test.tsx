import 'react-native';
import React from 'react';
import MenuItem from '../../app/components/MenuItem';
import {render, RenderAPI} from '@testing-library/react-native';
import {IApiMenuItem} from '../../app/api/Menu';

describe('MenuItem', () => {
  const item: IApiMenuItem = {
    name: 'My Menu Item',
    price: 123,
    description: 'My Menu Desc',
    url: '',
  };
  const getRender = (): RenderAPI => render(<MenuItem item={item} index={0} />);

  it('Render title', async () => {
    const element = await getRender().getByTestId('menu-item-title-0');
    expect(element).toBeTruthy();
    expect(element.children.join()).toEqual(item.name);
  });

  it('Render description', async () => {
    const element = await getRender().getByTestId('menu-item-description-0');
    expect(element).toBeTruthy();
    expect(element.children.join()).toEqual(item.description);
  });

  it('Render price', async () => {
    const element = await getRender().getByTestId('menu-item-price-0');
    expect(element).toBeTruthy();
    expect(element.children.join()).toEqual(`$ ${item.price}`);
  });

  it('Jest snapshot', async () => {
    const json = await getRender().toJSON();
    expect(json).toMatchSnapshot();
  });
});
