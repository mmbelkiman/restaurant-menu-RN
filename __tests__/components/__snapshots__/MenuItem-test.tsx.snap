// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem Jest snapshot 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "gray",
        "height": 1,
        "marginHorizontal": "10%",
        "opacity": 0.2,
      }
    }
  />
  <View
    style={
      Object {
        "padding": 30,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontFamily": "PlayfairDisplay-Medium",
          "fontSize": 23,
          "textAlign": "center",
        }
      }
      testID="menu-item-title-0"
    >
      My Menu Item
    </Text>
    <Text
      style={
        Object {
          "color": "gray",
          "fontSize": 18,
          "paddingTop": 10,
          "textAlign": "center",
        }
      }
      testID="menu-item-description-0"
    >
      My Menu Desc
    </Text>
    <Text
      style={
        Object {
          "color": "gray",
          "fontSize": 14,
          "paddingTop": 10,
          "textAlign": "center",
        }
      }
      testID="menu-item-price-0"
    >
      $ 123
    </Text>
  </View>
</View>
`;
